server/sublist.go

```diff
@@ -83,7 +83,7 @@ func NewSublist() *Sublist {
 // Insert adds a subscription into the sublist
 func (s *Sublist) Insert(sub *subscription) error {
 	// copy the subject since we hold this and this might be part of a large byte slice.
-	subject := string(append([]byte(nil), sub.subject...))
+	subject := string(sub.subject)
```

Comment:
Currently the compiler is not smart enough to recognize this and even with current Go tip it doesn't optimize this. I don't know if this is something that can be expected in the future.

What we could do is something like this:

```go
subject := (" " + string(sub.subject))[1:]
```

This forces an allocation, but doesn't need the second allocation.